Adding NN regressor support.

adam-ah opened the pull request.

adam-ah commented:
ping @jonnor

jonnor commented:
Hi @adam-ah, thank you for this merge request. It looks good in general. Making some review comments in.

jonnor commented:
Also, you should add yourself to the citation section in the README.md file

adam-ah commented:
hi @jonnor I addressed your comment, let me know if this is what you wanted.

I'll leave it up to you to update the citation/authors section, as Mark Cooke was listed as contributor too :)

jonnor commented on emlearn/net.py:
```diff
@@ -46,6 +52,9 @@ def predict_proba(self, X):
         classes = self.classifier.predict(X)
         return classes
+    
+    def regress(self, X):
```
@adam-ah This comment has not been addressed.

adam-ah replied:
@jonnor you might have forgotten to submit any of these comments beforehand - they just became visible to me (start review vs. add comment). I'll look through them

jonnor replied:
@adam-ah Whaat. Sorry about that! Long time since I did a review on Github (mostly using Gitlab professionally these days).

jonnor replied:
Props for getting 1/2 right from just my summary comment :D

jonnor reviewed (changes requested):
One change still requested, using predict instead of regress on Python interface. Rest looks good!

I am investigating the proper way to update Zenodo now.

jonnor commented on emlearn/net.py:
```diff
@@ -36,6 +37,11 @@ def __init__(self, activations, weights, biases, classifier):
             self.classifier = common.CompiledClassifier(code, name=name, call=func)
+        elif classifier == 'regressor':
+            name = 'mynet'
+            func = 'eml_net_regress1(&{}, values, length)'.format(name)
```
For later

jonnor commented:
Figured out how to update Zenodo - turns out one can use a CITATION.cff file in the repo - and they will automatically pick up the info on releases. Tested now by adding Mark - it looks to have worked, https://zenodo.org/record/8212812

jonnor reviewed (approved):
Looks good!
